### ISSUE TITLE
False positive with GuardClause and `raise`

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#9055]
Am I right in preferring

```ruby
# better:
def foo
  ...
  if some_long_condition
    raise some_long_error
  end
end

# worse
def foo
  ...
  return unless some_long_condition
    
  raise some_long_error
end
```

I'm assuming that `raise ... if ...` does not fit on one line, and the the body of the `if` is a single `raise` statement.

Here's the test that I think should pass but doesn't...

```ruby
  it 'does not register an offense for bodies that are simple `raise`' do
    expect_no_offenses(<<~'RUBY')
      def foo
        do_something
        if some_complex_condition_for_an_error?
          raise "Something complicated #{ what_went_wrong }; #{ what_to_do_instead }"
        end
      end
    RUBY
  end
```

**Comment by marcandre** [comment]
Bah, maybe it's not so bad, idk 🤷‍♂️
